### PR TITLE
Docs - Markdown proof of concept

### PR DESCRIPTION
**Pull request by samrichca**
Markdown POC:
- add basic Markdown processing
- add table of contents, in section & page variations
- add prev/next buttons
- use _contents.yaml for organization for now

Other changes:
- wrap all pages in `<Layout />` component
- update README
- add aliases for importing local files
- improve 404 page
- eslint/prettier formatting
- add ES6 modules in server-side Gatsby code
- convert class components to functional where possible
- changed how site builds in production - docs pages will be built but they'll all look like 404 pages. my strategy of not building them at all stopped working when I added GraphQL requests.

### CODE REVIEW

File: README-documentation.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Documentation section
```
Comment (samrichca) [inline README-documentation.md:1]: wrote a separate readme explaining how docs work

File: README.md [hunk 2]
```diff
@@ -1,10 +1,13 @@
-# Overture.bio Website
+# Overture.bio
 
-This site is hosted on Netlify. See **Deployment** for info.
+- Hosted on Netlify. See **Deployment** for info.
+- Based on the [Gatsby Starter Business](https://gatsby-starter-business.netlify.com).
 
-There are additional READMEs in inner folders.
+## Additional READMEs
```
Comment (samrichca) [inline README.md:6]: added links to other READMEs

File: gatsby-browser.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import React from 'react';
```
Comment (samrichca) [inline gatsby-browser.js:1]: wrapping all pages in <Layout /> component

File: gatsby-config.js [hunk 4]
```diff
@@ -38,12 +36,45 @@ module.exports = {
         exclude: [],
       },
     },
-
+    {
```
Comment (samrichca) [inline gatsby-config.js:39]: add markdown & yaml processing

File: gatsby-node.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+          // use for graphQL query in template
+          id,
+          section,
+        },
+      });
+    }
+  });
+}
+
+export function onCreateWebpackConfig({ actions }) {
```
Comment (samrichca) [inline gatsby-node.js:94]: adding aliases so it's easier to import things locally, e.g. `import { x } from 'components'`

File: gatsby-node.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
+// keep at top of file
+// require('dotenv').config({
+//   path: `.env.${process.env.NODE_ENV}`,
+// });
+
+const path = require('path');
+const startCase = require('lodash.startcase');
+
+export function onCreateNode({ node, getNode, actions: { createNodeField } }) {
```
Comment (samrichca) [inline gatsby-node.js:9]: importing markdown into gatsby graph & then making pages

File: gatsby-ssr.js [hunk 7]
```diff
@@ -0,0 +1,6 @@
+import React from 'react';
+import { Layout } from 'components';
```
Comment (samrichca) [inline gatsby-ssr.js:2]: wrapping all pages in <Layout /> component (has to be the same as gatsby-browser.js, at least in our implementation)

File: jsconfig.json [hunk 8]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (samrichca) [inline jsconfig.json:1]: this has to do with the webpack aliases

File: markdown/documentation/song/_contents.yaml [hunk 9]
```diff
@@ -0,0 +1,15 @@
+- title: Song
```
Comment (samrichca) [inline markdown/documentation/song/_contents.yaml:1]: the next several files are stub content for `/documentation/song`

File: meta/config.js [hunk 10]
```diff
@@ -1,16 +1,14 @@
-/**
- * Created  on 31/3/18
- */
 module.exports = {
```
Comment (samrichca) [inline meta/config.js:1]: formatting + added google analytics tracking ID here

File: meta/docs-schema.yaml [hunk 11]
```diff
@@ -0,0 +1,19 @@
+# this exists to stop graphQL from breaking
```
Comment (samrichca) [inline meta/docs-schema.yaml:1]: the site crashes if I query for 4 levels of nesting and there's only 2 available, so this is one solution: add mock data. there's also a type definition API that I can use with to make this more permanent, but I wasn't able to get it working quickly and this is OK for now.

File: package.json [hunk 12]
```diff
@@ -34,22 +42,24 @@
   "license": "AGPL-3.0-or-later",
   "main": "n/a",
   "scripts": {
+    "//": "ES6 modules are not currently supported in Gatsby so we have to use the esm library.",
     "build-prod": "netlify build --context production",
     "build-qa": "netlify build --context branch-deploy",
-    "build": "gatsby build --prefix-paths",
+    "build": "cross-env NODE_OPTIONS=\"-r esm\" gatsby build --prefix-paths",
```
Comment (samrichca) [inline package.json:48]: adding the ability to use es6 modules in gatsby-*.js files

File: src/components/BottomCallout/index.js [hunk 13]
```diff
@@ -1,21 +1,21 @@
-import React from 'react'
-import './styles.scss'
-import { Icon } from '../'
+import React from 'react';
```
Comment (samrichca) [inline src/components/BottomCallout/index.js:1]: the next run of files only has formatting & import alias changes.

File: src/components/TableOfContents/HeadingsTableOfContents/index.js [hunk 14]
```diff
@@ -0,0 +1,6 @@
+import React from 'react';
+import TableOfContents from '../TableOfContents';
```
Comment (samrichca) [inline src/components/TableOfContents/HeadingsTableOfContents/index.js:2]: created a simple recursive table of contents

File: src/components/index.js [hunk 15]
```diff
@@ -1,20 +1,23 @@
...
-export { default as GettingStarted } from './GettingStarted'
-export { default as GridFeature } from './GridFeature'
-export { default as ProgressBar } from './ProgressBar'
-export { ProductFeature, ProductFeatureRow } from './ProductFeature'
-export { default as NavBar } from './NavBar'
-export { default as ProductHero } from './ProductHero'
-export { default as Layout } from './Layout'
-export { H1, H2, H4 } from './Typography'
-export { default as Hero } from './Hero'
+export { BottomCallout, Callout } from './BottomCallout';
```
Comment (samrichca) [inline src/components/index.js:2]: improved importing/exporting of components. there were some components that couldn't be imported from this file.

File: src/pages/404.js [hunk 16]
```diff
@@ -1,27 +1,22 @@
...
-  <div>
-    <section className="hero is-primary is-bold is-large">
-      <div className="hero-body">
-        <div className="container">
-          <div className="columns">
-            <div className="column is-10 is-offset-1">
-              <div className="section is-centered">
-                <h1 className="title">404: NOT FOUND</h1>
-                <h2 className="subtitle">
-                  You just hit a route that doesn&#39;t exist... the sadness.
```
Comment (samrichca) [inline src/pages/404.js:16]: Changed this message

File: src/pages/documentation/index.js [hunk 17]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
```
Comment (samrichca) [inline src/pages/documentation/index.js:1]: added a stub documentation page

File: utils.js [hunk 18]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
```
Comment (samrichca) [inline utils.js:1]: gonna keep this because I'll probably need it

**Review by anncatton — APPROVED** [review]

**Pull request merged**